pkg/appauth/manager/json/json.go

```diff
@@ -17,3 +17,185 @@
 // or submit itself to any jurisdiction.
 
 package json
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"time"
+
+	apppb "github.com/cs3org/go-cs3apis/cs3/auth/applications/v1beta1"
+	authpb "github.com/cs3org/go-cs3apis/cs3/auth/provider/v1beta1"
+	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	typespb "github.com/cs3org/go-cs3apis/cs3/types/v1beta1"
+	"github.com/cs3org/reva/pkg/appauth"
+	"github.com/cs3org/reva/pkg/appauth/manager/registry"
+	"github.com/cs3org/reva/pkg/errtypes"
+	"github.com/cs3org/reva/pkg/user"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/sethvargo/go-password/password"
+)
+
+func init() {
+	registry.Register("json", New)
+}
+
+type config struct {
+	File          string `mapstructure:"file"`
+	TokenStrength int    `mapstructure:"token_strength"`
+}
+
+type jsonManager struct {
+	config *config
+	model  *jsonModel
+}
+
+type jsonModel struct {
+	file  string
+	State map[string]*apppb.AppPassword `json:"state"`
```

Comment:
Maybe call this Passwords?

Also, I think it would make more sense to have the user ID as the key in the map. One user can create multiple passwords, so that would make retrieval easier. And listing would then become an O(1) operation, instead of iterating over the whole list.